src/System.IO.FileSystem/src/System/IO/Win32FileSystemEnumerable.cs

```diff
@@ -217,7 +217,6 @@ private Win32FileSystemEnumerableIterator(string fullPath, string normalizedSear
 
             if (searchCriteria != null)
             {
-                PathInternal.CheckInvalidPathChars(fullPath);
```

Comment:
this is not the only use in corefx. remove the impl and test for checkinvalidpathchars as well.